client/src/commands/UserInputUtil.ts

```diff
@@ -204,26 +199,73 @@ export class UserInputUtil {
         return vscode.window.showQuickPick(peerNames, quickPickOptions);
     }
 
-    public static async showChaincodeAndVersionQuickPick(prompt: string, peers: Set<string>): Promise<IBlockchainQuickPickItem<{ chaincode: string, version: string }> | undefined> {
+    public static async showChaincodeAndVersionQuickPick(prompt: string, peers: Set<string>, packageLabel?: string): Promise<IBlockchainQuickPickItem<{ packageEntry: PackageRegistryEntry, workspace: vscode.WorkspaceFolder }> | undefined> {
         const fabricConnectionManager: FabricConnectionManager = FabricConnectionManager.instance();
         const connection: IFabricConnection = fabricConnectionManager.getConnection();
         if (!connection) {
             return Promise.reject('No connection to a blockchain found');
         }
 
-        const quickPickItems: Array<IBlockchainQuickPickItem<{ chaincode: string, version: string }>> = [];
+        // Get all installed smart contracts
+        let tempQuickPickItems: IBlockchainQuickPickItem<{ packageEntry: PackageRegistryEntry, workspace: vscode.WorkspaceFolder }>[] = await this.getInstalledContracts(connection, peers);
 
-        for (const peer of peers) {
-            const chaincodes: Map<string, Array<string>> = await connection.getInstalledChaincode(peer);
-            chaincodes.forEach((versions: string[], chaincodeName: string) => {
-                versions.forEach((version: string) => {
-                    const data: {chaincode: string, version: string} = {chaincode: chaincodeName, version: version};
-                    quickPickItems.push({label: `${chaincodeName}@${version}`, data: data});
-                });
-            });
+        // Get all packaged smart contracts
+        tempQuickPickItems = await this.getPackagedContracts(tempQuickPickItems);
```

Comment:
can't you just push the returned quickPick items to the master list rather than passing around the list?